capstone/src/main/java/com/google/sps/servlets/AssociationServlet.java

```diff
@@ -15,15 +23,44 @@
 public class AssociationServlet extends HttpServlet {
 
   private static final String OUTPUT_TYPE = "applications/json;";
-  private final ArrayList<String> positive = new ArrayList(Arrays.asList("hi", "test1", "test2"));
-  private final ArrayList<String> negative = new ArrayList(Arrays.asList("yeet", "hi", "think"));
+  public static final int LIMIT = 3;
+
+  private DatastoreService datastore;
+
+  public void init() throws ServletException {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(OUTPUT_TYPE);
 
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(LIMIT);
+
+    Query posQuery =
+        new Query(AssociationResult.ENTITY_KIND).addSort("score", SortDirection.DESCENDING);
+    ArrayList<String> positive = new ArrayList<String>();
+    extractContent(datastore.prepare(posQuery).asQueryResultList(fetchOptions), positive);
+
+    Query negQuery =
+        new Query(AssociationResult.ENTITY_KIND).addSort("score", SortDirection.ASCENDING);
+    ArrayList<String> negative = new ArrayList<String>();
+    extractContent(datastore.prepare(negQuery).asQueryResultList(fetchOptions), negative);
+
     AssociationData output = new AssociationData(positive, negative);
     Gson gson = new Gson();
     response.getWriter().println(gson.toJson(output));
   }
+
+  /**
+   * Adds the names of the associations to the arraylist passed in
+   *
+   * @param query the entities to get the names from
+   * @param output the arraylist to add the names of the entities to
+   */
+  private void extractContent(QueryResultList<Entity> query, ArrayList<String> output) {
+    for (Entity entity : query) {
+      output.add((String) entity.getProperty("name"));
+    }
+  }
```

Comment:
This is interesting that you're updating output by reference.

I think usually returning the output is clearer. What do you think?

Reply:
I think this is an artifact of worrying about possibly having to combine multiple arraylists. I agree that its clearer to just return it, I can swap that.